pkg/controller/pv.go

```diff
@@ -51,33 +56,61 @@ func (r *reconcilePV) Reconcile(ctx context.Context, request reconcile.Request)
 	var pv corev1.PersistentVolume
 	if err := r.client.Get(ctx, request.NamespacedName, &pv); err != nil {
 		if errors.IsNotFound(err) {
-			return reconcile.Result{}, err
+			return reconcile.Result{}, nil
 		}
 		klog.Errorf("failed to get PV (%s): %v", request.Name, err)
 		return reconcile.Result{Requeue: true}, err
 	}
 
-	if pv.Spec.CSI == nil || pv.Spec.CSI.Driver != DriverName {
+	if pv.Spec.CSI == nil || pv.Spec.CSI.Driver != azureutils.DriverName {
 		return reconcile.Result{}, nil
 	}
 
 	var azVolume v1alpha1.AzVolume
 	diskName, err := azureutils.GetDiskNameFromAzureManagedDiskURI(pv.Spec.CSI.VolumeHandle)
+	azVolumeName := strings.ToLower(diskName)
 	if err != nil {
 		klog.Errorf("failed to extract proper diskName from pv(%s)'s volume handle (%s): %v", pv.Name, pv.Spec.CSI.VolumeHandle, err)
 		// if disk name cannot be extracted from volumehandle, there is no point of requeueing
 		return reconcile.Result{}, err
 	}
-	if err := r.client.Get(ctx, types.NamespacedName{Namespace: r.namespace, Name: diskName}, &azVolume); err != nil {
-		// if underlying PV was found but AzVolume was not. Might be due to stale cache
-		klog.V(5).Infof("failed to get AzVolume (%s): %v", diskName, err)
+	if err := r.client.Get(ctx, types.NamespacedName{Namespace: r.namespace, Name: azVolumeName}, &azVolume); err != nil {
+		// if underlying PV was found but AzVolume was not. Might be due to stale cache, so try until found, or maxTry Reached
+		if !errors.IsNotFound(err) {
+			klog.V(5).Infof("failed to get AzVolume (%s): %v", diskName, err)
+			return reconcile.Result{Requeue: true}, err
+		}
+		r.retryMutex.RLock()
+		numRetry, ok := r.retryMap[azVolumeName]
+		r.retryMutex.RUnlock()
+		if !ok {
+			r.retryMutex.Lock()
+			if _, ok = r.retryMap[azVolumeName]; !ok {
```

Comment:
If the element exists, we want to use the current value rather than 1 on line 92 below, so this should be:
~~~
if numRetry, ok = r.retryMap[azVolumeName]; !ok {
    r.retryMap[azVolumeName] = 1
    numRetry = 1
}
~~~

Reply:
Ah, good catch. I forgot to move the numTry update into the second check.